app/lib/package/backend.dart

```diff
@@ -524,6 +525,22 @@ class PackageBackend {
               'The `environment` field has invalid characters.');
         }
       }
+      if (googleCloud != null) {
+        final isEnabled = googleCloud.isEnabled ?? false;
+        // normalize input values
+        final serviceAccountEmail =
+            googleCloud.serviceAccountEmail?.trim() ?? '';
+        googleCloud.serviceAccountEmail = serviceAccountEmail;
+
+        InvalidInputException.check(
+            !isEnabled || serviceAccountEmail.isNotEmpty,
+            'The service account email field must not be empty when enabled.');
+
+        if (serviceAccountEmail.isNotEmpty) {
+          InvalidInputException.check(isValidEmail(serviceAccountEmail),
+              'The service account email is not valid: `$serviceAccountEmail`.');
```

Comment:
Let's require that they end with `.gserviceaccount.com`, see:
https://cloud.google.com/iam/docs/service-accounts#types

We can augment such an error message to say that: you can file a bug if your service account doesn't end with `.gserviceaccount.com`.